### DISCUSSION TITLE
Topic re-subscription required on app update?

### DISCUSSION THREAD

**Discussion by JoeToeniskoetter** [invertase/react-native-firebase#4940] [Q&A]
Quick question here. I have an application that allows users to subscribe to topic based notifications. If I push out an update to my app on the app store and my users download it, will those subscriptions survive the app update? Or do I need to manually resubscribe the users to the topics?

Thanks,
Joe

**Comment by mikehardy** [comment]
I *believe* they are based on device tokens so they should survive unless the installation id (from firebase installations package) changes as it is used as the token if I recall correctly

However the only way to be sure is to have a non-production app that you test with

**Reply by JoeToeniskoetter** [reply]
Thanks Mike. This brings up something else I didn't think about. If a token becomes invalid while a user is using my app, I will need to resubscribe them. I'm assuming this is the reason behind using `messaging().onTokenRefresh()` listener?

One last question. Since I am storing a user's subscriptions in AsyncStorage, do you see any downfall to me looping through these subscriptions and calling `messaging().subscribeToTopic()` on each app startup?

**Reply by mikehardy** [reply]
I'm sorry, but I haven't used topics much personally so I don't have direct experience, yet in my experience messaging stuff has to be tested tested tested so I hesitate to say anything else, I'm just not sure enough to offer useful direction. Perhaps someone else is - otherwise I'd post on stackoverflow there is a reasonably lively group of firebase people there and this is a general firebase question, not react-native specific even
